### PR TITLE
Adds missing anonymous_auth_enabled parameter

### LINKED ISSUE #26668: Datasource aws_opensearch_domain fails when advanced_security_options is enabled
### Terraform CLI and Terraform AWS Provider Version

```
Terraform v1.2.8
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v4.29.0
```

### Affected Resource(s)

* Datasource `aws_opensearch_domain`

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "4.29.0"
    }
  }
}

provider "aws" {
  region = "eu-west-3"
}

resource "aws_opensearch_domain" "bug" {
  domain_name = "bug"

  cluster_config {
    instance_type            = "t3.small.search"
    instance_count           = 1
    dedicated_master_enabled = false
  }
  ebs_options {
    ebs_enabled = true
    volume_size = 10
    volume_type = "gp2"
  }
  node_to_node_encryption {
    enabled = true
  }
  encrypt_at_rest {
    enabled = true
  }
  domain_endpoint_options {
    enforce_https       = true
    tls_security_policy = "Policy-Min-TLS-1-2-2019-07"
  }
  advanced_security_options {
    enabled                        = true
    internal_user_database_enabled = true
    master_user_options {
      master_user_name     = "elastic"
      master_user_password = "Eel@st1c"
    }
  }
}

data "aws_opensearch_domain" "my_domain" {
  domain_name = aws_opensearch_domain.bug.domain_name
}

```

### Debug Output
```
...
aws_opensearch_domain.bug: Creation complete after 16m16s [id=arn:aws:es:eu-west-3:560417158718:domain/bug]
data.aws_opensearch_domain.my_domain: Reading...
╷
│ Error: error setting advanced_security_options: Invalid address to set: []string{"advanced_security_options", "0", "anonymous_auth_enabled"}
│ 
│   with data.aws_opensearch_domain.my_domain,
│   on main.tf line 47, in data "aws_opensearch_domain" "my_domain":
│   47: data "aws_opensearch_domain" "my_domain" {
│ 
╵
```

[See full log in attachment](https://github.com/hashicorp/terraform-provider-aws/files/9498573/terraform.log)

### References

**This configuration is perfectly working with provider version `4.28.0`**

**Maybe related [to this PR](https://github.com/hashicorp/terraform-provider-aws/pull/26503)?**

### PR DESCRIPTION
**Pull request by dpattmann**
### Description

The function ```flattenAdvancedSecurityOptions```, used from the ```aws_opensearch_domain``` resource, tries to set ```anonymous_auth_enabled``` to a boolean value but the field is missing in the data resource model. This PR adds this field to the data resource.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #26668

Output from acceptance testing:

```
$ make testacc TESTS=TestAccOpenSearchDomainDataSource_ PKG=opensearch
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/opensearch/... -v -count 1 -parallel 20  -run=TestAccOpenSearchDomainDataSource_ -timeout 180m
=== RUN   TestAccOpenSearchDomainDataSource_Data_basic
=== PAUSE TestAccOpenSearchDomainDataSource_Data_basic
=== RUN   TestAccOpenSearchDomainDataSource_Data_advanced
=== PAUSE TestAccOpenSearchDomainDataSource_Data_advanced
=== CONT  TestAccOpenSearchDomainDataSource_Data_basic
=== CONT  TestAccOpenSearchDomainDataSource_Data_advanced
--- PASS: TestAccOpenSearchDomainDataSource_Data_basic (1597.77s)
--- PASS: TestAccOpenSearchDomainDataSource_Data_advanced (1856.79s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/opensearch	1858.491s
...
```

### CODE REVIEW

**Comment by fmartinou** [comment]
Hi,

Any news regarding this PR?
Is something missing to get it merged and available in the next release?

File: .changelog/26746.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+data/aws_opensearch_domain: Adds missing anonymous_auth_enabled parameter to aws_opensearch_domain data resource
```
Comment (sftim) [inline .changelog/26746.txt:2]:
```suggestion
data/aws_opensearch_domain: Add missing `anonymous_auth_enabled` parameter to `aws_opensearch_domain` data resource
```

**Comment by anithapriyanatarajan** [comment]
Hi, can someone please confirm if this issue is fixed?

**Comment by neovatar** [comment]
We encountered the same problem and are stuck with an old provider version - a merge and release of the PR would really be appreciated!

**Comment by ltdeoliveira** [comment]
Any ETA for merging this PR? This fixes a critical issue for my team.

**Comment by kai-ten** [comment]
Joining the noise, would love to see this old problem get put to bed... thanks

**Comment by barthel** [comment]
Thanks @ewbankkit for take over and pushing this issue.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccOpenSearchDomainDataSource_Data_advanced' PKG=opensearch ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/opensearch/... -v -count 1 -parallel 3  -run=TestAccOpenSearchDomainDataSource_Data_advanced -timeout 180m
=== RUN   TestAccOpenSearchDomainDataSource_Data_advanced
=== PAUSE TestAccOpenSearchDomainDataSource_Data_advanced
=== CONT  TestAccOpenSearchDomainDataSource_Data_advanced
--- PASS: TestAccOpenSearchDomainDataSource_Data_advanced (2452.31s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/opensearch	2457.830s
```

**Comment by ewbankkit** [comment]
@dpattmann Thanks for the contribution :tada: :clap:.

**Pull request merged**